libbpfgo/libbpfgo.go

```diff
@@ -478,6 +478,67 @@ func (b *BPFMap) Update(key, value interface{}) error {
 	return nil
 }
 
+type BPFMapIterator struct {
+	b *BPFMap
+
+	err error
+
+	keySize int
+
+	prev []byte
+	next []byte
+}
+
+func (b *BPFMap) Iter(keySize int) *BPFMapIterator {
+	return &BPFMapIterator{
+		b:       b,
+		keySize: keySize,
+		prev:    nil,
+		next:    nil,
+	}
+}
+
+func (it *BPFMapIterator) Next() bool {
+	if it.err != nil {
+		return false
+	}
+
+	prevPtr := unsafe.Pointer(nil)
+	if it.next != nil {
+		prevPtr = unsafe.Pointer(&it.next[0])
+	}
+
+	next := make([]byte, it.keySize)
+	nextPtr := unsafe.Pointer(&next[0])
+
+	errC := C.bpf_map_get_next_key(it.b.fd, prevPtr, nextPtr)
+	if errC == -1 {
+		it.prev = it.next
+		it.next = next
+		return false
+	}
+	if errC != 0 {
```

Comment:
According to the man page, there are only two possible return values: 0 or -1.
If so, how can this line ever evaluate to true (since we already checked for `errC == -1` above)?